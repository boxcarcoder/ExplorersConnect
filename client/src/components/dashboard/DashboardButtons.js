import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';

const DashboardButtons = () => {
  return (
    <Fragment>
      <div className='buttons vert-m-1'>
        <Link to='/edit-profile' className='btn'>
          <i className='fas fa-user-circle text-primary'></i> Edit profile
        </Link>
        <Link to='/add-destination' className='btn'>
          <i className='fas fa-map-signs text-primary'></i> Add Destinations
        </Link>
        <Link to='/add-gear' className='btn'>
          <i className='fas fa-snowboarding text-primary'></i> Add Gear
        </Link>
      </div>
    </Fragment>
  );
};

export default DashboardButtons;
