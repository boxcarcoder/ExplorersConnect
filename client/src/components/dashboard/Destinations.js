import React, { Fragment } from 'react';

//redux
import { PropTypes } from 'prop-types';

const Destinations = ({ destinations }) => {
  // Use the profile.destinations state (from Dashboard component) for destinations data.
  const allDestinations = destinations.map(destination => (
    <tr key={destination._id}>
      <td>{destination.hikingTrails}</td>
      <td>{destination.campSites}</td>
      <td>{destination.waterAreas}</td>
      <td>{destination.slopes}</td>
      <td>{destination.crags}</td>
    </tr>
  ));

  return (
    <Fragment>
      <h2 className='vert-m-1'>Explored Destinations</h2>
      <table className='table'>
        <thead>
          <tr>
            <th>Hiking</th>
            <th>Camping</th>
            <th>Water Sports</th>
            <th>Snow Sports</th>
            <th>Rock Climbing</th>
          </tr>
        </thead>
        <tbody>
          {allDestinations}
          <tr>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
          </tr>
        </tbody>
      </table>
    </Fragment>
  );
};

Destinations.propTypes = {
  destinations: PropTypes.array.isRequired
};

export default Destinations;
