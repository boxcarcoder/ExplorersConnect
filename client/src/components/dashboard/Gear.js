import React, { Fragment } from 'react';

const Gear = () => {
  return (
    <Fragment>
      <h2 className='vert-m-1'>Essential Gear</h2>
      <table className='table'>
        <thead>
          <tr>
            <th>Hiking</th>
            <th>Camping</th>
            <th>Water Sports</th>
            <th>Snow Sports</th>
            <th>Rock Climbing</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Salomon Odyssey Pro</td>
            <td>NatureHike CloudUp 2</td>
            <td>Chaco Z1</td>
            <td>Burton Board</td>
            <td></td>
          </tr>
          <tr>
            <td>Black Diamond Distance Z</td>
            <td>Aegismax M2</td>
            <td></td>
            <td></td>
            <td></td>
          </tr>
          <tr>
            <td></td>
            <td>Thermarest Prolite Plus</td>
            <td></td>
            <td></td>
            <td></td>
          </tr>
          <tr>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
          </tr>
        </tbody>
      </table>
    </Fragment>
  );
};

export default Gear;
