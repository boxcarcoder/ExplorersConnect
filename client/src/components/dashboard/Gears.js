import React, { Fragment } from 'react';

//redux
import { PropTypes } from 'prop-types';

const Gears = ({ gears }) => {
  // Use the profile.gears state (from Dashboard component) for gears data.
  const allGears = gears.map(gear => (
    <tr key={gear._id}>
      <td>{gear.hikeGear}</td>
      <td>{gear.campGear}</td>
      <td>{gear.waterGear}</td>
      <td>{gear.snowGear}</td>
      <td>{gear.rockClimbingGear}</td>
    </tr>
  ));

  return (
    <Fragment>
      <h2 className='vert-m-1'>Essential Gear</h2>
      <table className='table'>
        <thead>
          <tr>
            <th>Hiking</th>
            <th>Camping</th>
            <th>Water Sports</th>
            <th>Snow Sports</th>
            <th>Rock Climbing</th>
          </tr>
        </thead>
        <tbody>
          {allGears}
          <tr>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
            <td>
              <button className='btn btn-danger'>Delete</button>
            </td>
          </tr>
        </tbody>
      </table>
    </Fragment>
  );
};

Gears.propTypes = {
  gears: PropTypes.array.isRequired
};

export default Gears;
