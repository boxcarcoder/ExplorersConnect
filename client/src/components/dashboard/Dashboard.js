import React, { Fragment } from 'react';
import Spinner from '../layout/Spinner';
import { Link } from 'react-router-dom';
import DashboardButtons from './DashboardButtons';
import Destinations from './Destinations';
import Gears from './Gears';

//redux
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getCurrentProfile } from '../../actions/profile';

export const Dashboard = ({
  getCurrentProfile,
  profileState: { profile, loading },
}) => {
  //on the dashboard's first load, retrieve the logged in user's profile and save it into the profile redux state
  React.useEffect(() => {
    getCurrentProfile();
  }, []); //*** */

  if (loading) {
    return <Spinner />;
  } else {
    return (
      <Fragment>
        {/*check if the logged in user has a profile*/}
        {profile !== null ? (
          <Fragment>
            <h1 className='large text-primary'>Dashboard</h1>
            <p>
              <i className='fas fa-user'></i>Welcome {profile.user.name}
            </p>
            <DashboardButtons />
            <Destinations destinations={profile.destinations} />
            <Gears gears={profile.gears} />
          </Fragment>
        ) : (
          <Fragment>
            <p>You have not set up a profile yet!</p>
            <Link to='/create-profile' className='btn btn-primary vert-m-1'>
              Create Profile
            </Link>
          </Fragment>
        )}
      </Fragment>
    );
  }
};

Dashboard.propTypes = {
  getCurrentProfile: PropTypes.func.isRequired,
  profileState: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  profileState: state.profile,
});

export default connect(mapStateToProps, { getCurrentProfile })(Dashboard);
