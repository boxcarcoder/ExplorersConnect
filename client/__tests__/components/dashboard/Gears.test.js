import React from 'react';
import { shallow } from 'enzyme';

// Import the non-connected component
import { Gears } from '../../../src/components/dashboard/Gears';

// Globals for tests
let wrapper;
let props;
let mockDelete = jest.fn();

describe('<Gears /> component.', () => {
  describe('Successfully renders', () => {
    test('with 0 gears.', () => {
      props = {
        gears: [],
        deleteGears: mockDelete,
      };

      wrapper = shallow(<Gears {...props} />);

      // Execute the test.
      const firstGear = wrapper.find('td');
      expect(firstGear.exists()).toBe(false);
    });

    test('with existing gears.', () => {
      props = {
        gears: [
          {
            hikeGear: 'test hike gear 2',
            campGear: 'test camp gear 1',
          },
          {
            hikeGear: 'test hike gear 1',
          },
        ],
        deleteGears: mockDelete,
      };

      wrapper = shallow(<Gears {...props} />);

      // Execute the test.
      // Create snapshot of the DOM render.
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('If there are existing gears,', () => {
    beforeEach(() => {
      props = {
        gears: [
          {
            hikeGear: 'test hike gear 2',
            campGear: 'test camp gear 1',
          },
          {
            hikeGear: 'test hike gear 1',
          },
        ],
        deleteGears: mockDelete,
      };

      wrapper = shallow(<Gears {...props} />);
    });

    test('there are buttons to delete gears.', () => {
      const deleteButton = wrapper.find('button');

      // Execute the test.
      expect(deleteButton.exists()).toBe(true);
    });

    test('clicking the delete button triggers the deleteGears() action.', () => {
      const deleteButton = wrapper.find('button').at(0);

      deleteButton.simulate('click');
      expect(mockDelete).toHaveBeenCalled();
    });
  });
});
