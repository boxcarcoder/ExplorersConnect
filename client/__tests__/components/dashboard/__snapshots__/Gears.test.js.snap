// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gears /> component. Successfully renders with existing gears. 1`] = `
<Fragment>
  <h2
    className="vert-m-1"
  >
    Essential Gear
  </h2>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Hiking
        </th>
        <th>
          Camping
        </th>
        <th>
          Water Sports
        </th>
        <th>
          Snow Sports
        </th>
        <th>
          Rock Climbing
        </th>
        <th>
          Remove?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test hike gear 2
        </td>
        <td>
          test camp gear 1
        </td>
        <td />
        <td />
        <td />
        <td>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          test hike gear 1
        </td>
        <td />
        <td />
        <td />
        <td />
        <td>
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>
`;
